### PR TITLE
Add high-level events

### PR DESCRIPTION
**Pull request by saul**
Adds some high-level events. These events are likely to be used a lot and expecting everyone to hand-roll them is a poor experience. Adds a framework for supplemental events that has no overhead on parse time, unless the events are actually listened on.

- `DemoFile#molotovDetonate`
- `DemoFile#grenadeTrajectory`
- `DemoFile#itemPurchase`

**Todo**

- [x] Update readme
- [x] Add integration tests for new events

### CODE REVIEW

**Comment by n1c** [comment]
This is an awesome idea!

I don't think it's possible - but could these events also bundle in the data for throwing player etc? Currently hand-rolling that stuff is also quite painful.

**Comment by saul** [comment]
Thanks - yes it's definitely possible. Both of these events have a `thrower` property.

File: src/demo.ts [hunk 1]
```diff
@@ -432,7 +472,7 @@ export class DemoFile extends EventEmitter {
   /**
    * When parsing, set to current tick.
    */
-  public currentTick: number = 0;
+  public currentTick: number = -1;
```
Comment (saul) [inline src/demo.ts:475]: - [ ] Needs release note

File: src/entities.ts [hunk 2]
```diff
@@ -735,17 +756,7 @@ export class Entities extends EventEmitter {
...
-    // eslint-disable-next-line no-restricted-syntax
-    for (const tableName in this.tableClassMap) {
-      if (baseline[tableName]) {
-        // UNSAFE: we know this table exists in `tableClassMap`
-        klass = this.tableClassMap[tableName]!;
-        break;
-      }
-    }
-
+    const klass = this._serverClassConstructor.get(classId) || Networkable;
```
Comment (saul) [inline src/entities.ts:759]: - [ ] Bug fix. Needs release note

**Pull request merged**